py-polars/Makefile

```diff
@@ -67,7 +68,8 @@ cleaner: clean  ## clean + remove VENV and other auto-generated dirs & files
 	-rm -f polars/polars.abi3.so
 	-find . \( -type f -name '*.py[co]' \) -or \( -type d -name __pycache__ \) -delete
```

Comment:
This leads to errors on my machine (WSL2 Ubuntu 22.04):
```
find: cannot delete ‘./tests/unit/__pycache__’: Directory not empty
find: cannot delete ‘./tests/unit/io/__pycache__’: Directory not empty
find: cannot delete ‘./polars/__pycache__’: Directory not empty
find: cannot delete ‘./polars/internals/__pycache__’: Directory not empty
find: cannot delete ‘./polars/internals/lazyframe/__pycache__’: Directory not empty
find: cannot delete ‘./polars/internals/sql/__pycache__’: Directory not empty
find: cannot delete ‘./polars/internals/series/__pycache__’: Directory not empty
find: cannot delete ‘./polars/internals/expr/__pycache__’: Directory not empty
find: cannot delete ‘./polars/internals/dataframe/__pycache__’: Directory not empty
```

The old command works fine:
```find -type f -name '*.py[co]' -delete -or -type d -name __pycache__ -delete```

Reply:
I do not have access to the windows machine (or WSL2). Hence, guessing...
Could you please try replacing the `find` line with the following one (add proper makefile TAB indentation):
```
-find . \( -type f -name '*.py[co]' -or -type d -name __pycache__ \) -delete
```
This hopefully works on all platforms. 
Running `find` line you suggested, the one without any `\(...\)` parenthesis, does not delete `__pycache__` directories at least on MacOS-12.6.1.